Update Node to v12

AJMiller opened the pull request:
[Pivotal Tracker](https://www.pivotaltracker.com/story/show/172580219)

Also:

- runs 'npm update' to get latest package updates
- Updates version numbers in setup docs

Commit:
Update Node to v 12

- runs 'npm update' to get latest package updates
- Updates version numbers in setup docs

bmack123 commented on SETUP.md:
```diff
@@ -4,8 +4,8 @@ The back-end is a full Rails project, but the expectation is that you will set u
 
-1. Install the project Ruby version `2.5.1`
-2. Install Node 8 or higher (the project expects `8.16.0`)
+1. Install the project Ruby version `2.6.5`
```
I realized I didn't notice this in the last project, glad you saw it here too/

bmack123 commented on SETUP.md:
```diff
@@ -4,8 +4,8 @@ The back-end is a full Rails project, but the expectation is that you will set u
-1. Install the project Ruby version `2.5.1`
-2. Install Node 8 or higher (the project expects `8.16.0`)
+1. Install the project Ruby version `2.6.5`
+2. Install Node 12 or higher (the project expects `12.16.2`)
```
We should really change this to just `install Node 16.12.2` as the package.json will complain if you don't install the exact version.

bmack123 reviewed (commented):
small change to setup. other than that looks good.

bmack123 commented on client/package.json:
```diff
@@ -1,17 +1,17 @@
-    "react": "^16.13.0",
-    "react-dom": "^16.13.0",
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1",
     "react-router-dom": "^5.1.2",
-    "react-scripts": "^3.4.0"
+    "react-scripts": "^3.4.1"
```
Were these done manually or automatically? There shouldn't be a difference between what resolves for these changes in the package-lock.json.

AJMiller replied:
I honestly don't remember updating them, but I reverted them just in case. I ran `npm update` and the lock file didn't change so I think we're good to leave them at the `.0` versions.

Commit:
Revert package.json patch version updates

These shouldn't affect the lock file so no need to update them.

Commit:
Clarify we need exactly node 12.16.2

The app will complain if the exact node version we specify isn't installed, so we should make that clear here.

bmack123 reviewed (approved):
LGTM

bmack123 commented:
Acceptance Testing
---
- [ ] The app still runs using node 12
